packages/adapters/ethers/src/client.ts

```diff
@@ -175,889 +95,394 @@ export class EthersAdapter {
-  }> {
-    return new Promise(resolve => {
-      const walletId = SafeLocalStorage.getItem(SafeLocalStorageKeys.WALLET_ID)
 
-      if (!walletId) {
-        throw new Error('No wallet id found to get approved networks data')
-      }
-
-      const providerConfigs = {
-        [ConstantsUtil.AUTH_CONNECTOR_ID]: {
-          supportsAllNetworks: true,
-          approvedCaipNetworkIds: PresetsUtil.WalletConnectRpcChainIds.map(
-            id => `${ConstantsUtil.EIP155}:${id}`
-          ) as CaipNetworkId[]
-        }
-      }
-
-      const networkData = providerConfigs[walletId as unknown as keyof typeof providerConfigs]
+    const connectors = Object.keys(this.ethersConfig || {}).filter(
+      key => key !== 'metadata' && key !== 'EIP6963'
+    )
 
-      if (networkData) {
-        resolve(networkData)
-      } else {
-        resolve({
-          supportsAllNetworks: true,
-          approvedCaipNetworkIds: []
+    connectors.forEach(connector => {
+      const key = connector === 'coinbase' ? 'coinbaseWalletSDK' : connector
+      if (this.namespace) {
+        this.addConnector({
+          id: connector,
+          explorerId: PresetsUtil.ConnectorExplorerIds[key],
+          imageUrl: options?.connectorImages?.[key],
+          name: PresetsUtil.ConnectorNamesMap[key],
+          imageId: PresetsUtil.ConnectorImageIds[key],
+          type: PresetsUtil.ConnectorTypesMap[key] ?? 'EXTERNAL',
+          info: { rdns: key },
+          chain: this.namespace,
+          chains: [],
+          provider: this.ethersConfig?.[connector as keyof ProviderType] as Provider
         })
       }
     })
   }
 
-  /**
-   * Checks the active providers and sets the provider. We call this when we initialize the adapter.
-   * @param config - The provider config
-   */
-  private checkActiveProviders(config: ProviderType) {
-    const walletId = SafeLocalStorage.getItem(SafeLocalStorageKeys.WALLET_ID)
-    const walletName = SafeLocalStorage.getItem(SafeLocalStorageKeys.WALLET_NAME)
+  public async connectWalletConnect(onUri: (uri: string) => void) {
+    const connector = this.connectors.find(c => c.type === 'WALLET_CONNECT')
 
-    if (!walletId) {
-      return
-    }
+    const provider = connector?.provider as UniversalProvider
 
-    const providerConfigs = {
-      [ConstantsUtil.INJECTED_CONNECTOR_ID]: {
-        provider: config.injected
-      },
-      [ConstantsUtil.COINBASE_SDK_CONNECTOR_ID]: {
-        provider: config.coinbase as unknown as ExternalProvider
-      },
-      [ConstantsUtil.EIP6963_CONNECTOR_ID]: {
-        provider: this.EIP6963Providers.find(p => p.info.name === walletName)?.provider
-      }
+    if (!this.caipNetworks || !provider) {
+      throw new Error(
+        'UniversalAdapter:connectWalletConnect - caipNetworks or provider is undefined'
+      )
     }
 
-    const activeConfig = providerConfigs[walletId as unknown as keyof typeof providerConfigs]
+    provider.on('display_uri', (uri: string) => {
```

Comment:
shouldn't this be done on the universal adapter level ? are we not using universal adapter here anymore?

Reply:
If you look at the `createAdapters` you see that if there is an blueprint, we will create the adapter for it with the specific namespace. If there are no adapters, we will create an UniversalAdapter instance for this namespace. All adapters will be using the same UniversalProvider instance.

We decided to go this way with Felipe, so that everything will be treated in the same way and we're not overcomplicating the codebase (wagmi issues, too much if else statements, etc)